src/aktualizr_lite/main.cc

```diff
@@ -0,0 +1,211 @@
+  OstreeDeployment *deployment = ostree_sysroot_get_booted_deployment(sysroot_smart.get());
+  if (deployment == nullptr) {
+    LOG_INFO << "No active deployment found";
+  } else {
+    LOG_INFO << "Active image is: " << ostree_deployment_get_csum(deployment);
+  }
+  return 0;
+}
+
+static int list_main(Config &config, const bpo::variables_map &unused) {
+  (void)unused;
+  auto storage = INvStorage::newStorage(config.storage);
+  auto client = SotaUptaneClient::newDefaultClient(config, storage);
+  Uptane::HardwareIdentifier hwid(config.provision.primary_ecu_hardware_id);
+
+  LOG_INFO << "Refreshing target metadata";
+  if (!client->updateImagesMeta()) {
+    LOG_ERROR << "Unable to update latest metadata, using local copy";
+  }
+
+  LOG_INFO << "Updates for available to " << hwid << ":";
+  for (auto const &t : client->allTargets()) {
+    for (auto const &it : t.hardwareIds()) {
+      if (it == hwid) {
+        LOG_INFO << t.custom_version() << "\tsha256:" << t.sha256Hash();
+        break;
+      }
+    }
+  }
+  return 0;
+}
+
+static std::unique_ptr<Uptane::Target> find_target(const std::shared_ptr<SotaUptaneClient> &client,
+                                                   Uptane::HardwareIdentifier &hwid, const std::string &version) {
+  std::unique_ptr<Uptane::Target> rv;
+  if (!client->updateImagesMeta()) {
+    LOG_ERROR << "Unable to update latest metadata, using local copy";
+  }
+  for (auto const &t : client->allTargets()) {
+    for (auto const &it : t.hardwareIds()) {
+      if (it == hwid) {
+        if (version == "latest" || version == t.custom_version()) {
+          return std_::make_unique<Uptane::Target>(t);
+        }
+      }
+    }
+  }
+  throw std::runtime_error("Unable to find update");
+}
+
+static int update_main(Config &config, const bpo::variables_map &variables_map) {
+  auto storage = INvStorage::newStorage(config.storage);
+  auto client = SotaUptaneClient::newDefaultClient(config, storage);
+  Uptane::HardwareIdentifier hwid(config.provision.primary_ecu_hardware_id);
+
+  std::string version("latest");
+  if (variables_map.count("update-name") > 0) {
+    version = variables_map["update-name"].as<std::string>();
+  }
+  LOG_INFO << "Finding " << version << " to update to...";
+  auto target = find_target(client, hwid, version);
+  LOG_INFO << "Updating to: " << *target;
+
+  std::vector<Uptane::Target> targets{*target};
+  auto result = client->downloadImages(targets);
+  if (result.status != result::DownloadStatus::kSuccess &&
+      result.status != result::DownloadStatus::kNothingToDownload) {
+    LOG_ERROR << "Unable to download update: " + result.message;
+    return 1;
+  }
+  auto iresult = client->PackageInstall(*target);
+  if (iresult.result_code.num_code != data::ResultCode::Numeric::kOk &&
+      iresult.result_code.num_code != data::ResultCode::Numeric::kNeedCompletion) {
+    LOG_ERROR << "Unable to install update: " << iresult.description;
+    return 1;
+  }
```

Comment:
Do you want to say anything about requiring a reboot if the result code is `kNeedCompletion`? We recently changed the code to fix a longstanding issue that an OSTree installation wasn't really complete until the system had restarted. We are now trying to make this much more clear, and you might want to do the same here.